src/utils/utils.cairo

```diff
@@ -780,15 +780,17 @@ namespace Helpers {
     // @return n the resulting array
     func bytes_to_bytes4_array{range_check_ptr}(
         data_len: felt, data: felt*, n_len: felt, n: felt*
-    ) -> (n_len: felt, n: felt*) {
+    ) -> (n_len: felt, n: felt*, last: felt, last_num_bytes: felt) {
```

Comment:
I think we should have a second function, `bytes_to_bytes4_array_partial` for this specific purpose

Reply:
As discussed IRL: this function is only once in the codebase for the PrecompileSHA256.
Updated the docstring